src/AutoRest.CSharp/Common/Generation/Writers/ClientOptionsWriter.cs

```diff
@@ -50,6 +51,21 @@ public static void WriteClientOptions(CodeWriter writer, ClientOptionsTypeProvid
                             }
                         }
                     }
+
+                    if (!Configuration.KeepOptionalClientParametersInConstructor)
+                    {
+                        if (clientOptions.ApiVersions is not null)
+                        {
+                            writer.Line();
+                        }
+
+                        foreach (var parameter in clientOptions.AdditionalParameters)
+                        {
+                            writer.WriteXmlDocumentationSummary(parameter.Description);
+                            writer.Line($"public {parameter.Type} {parameter.Name.FirstCharToUpperCase()} {{ get; set; }}");
```

Comment:
```suggestion
                            writer.Line($"public {parameter.Type} {parameter.Name.ToCleanName()} {{ get; set; }}");
```
well I would suggest we construct something like `Property` but we currently do not have a class to define a property. We only have `ObjectTypeProperty`. This could be future work to refactor.

Reply:
I created an issue https://github.com/Azure/autorest.csharp/issues/3940, since this PR changes API so it is better to get merged first.